dashboard/state_aggregator.py

```diff
@@ -241,11 +247,151 @@ def sort_func(entry):
         result.sort(key=sort_func, reverse=True)
         return list(islice(result, option.limit))
 
+    async def get_resource_summary(self, per_node: bool = False) -> dict:
+        """Summarizes the total/available resources in the cluster.
+
+        Returns:
+            if per_node == True:
+                {node_id -> resource_summary_dict}
+            else:
+                resource_summary_dict
+            resource_summary_dict's schema is in ResourceSummary
+        """
+        resources_available = {}
+        resources_total = {}
+
+        def fill_resources(resources: dict, data_field: str, node_list: list):
+            for node in node_list:
+                data = self._message_to_dict(
+                    message=node,
+                    fields_to_decode=["node_id"],
+                )
+                if per_node:
+                    resources[data["node_id"]] = data[data_field]
+                else:
+                    for k, v in data[data_field].items():
+                        if k in resources:
+                            resources[k] += v
+                        else:
+                            resources[k] = v
+
+        async def fill_available_resources():
+            reply = await self.data_source_client.get_all_available_resources(
+                timeout=DEFAULT_RPC_TIMEOUT
+            )
+            fill_resources(
+                resources_available, "resources_available", reply.resources_list
+            )
+
+        async def fill_total_resources():
+            reply = await self.data_source_client.get_all_node_info(
```

Comment:
Do we need 2 APIs here? Is there no API that can return both at the same time in GCS?